src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+  }
+
+  bite(someone) {
+    if (!someone.hidden && !(someone instanceof Carnivore)) {
```

Comment:
You should check if a bite target is a Herbivoire. Change the logic, please.

> Carnivore has a `bite` method, which takes a herbivore object and decreases the object's health by 50.

Reply:
I understood the condition but didn't understand why Carnivore can't bite any Animals...

Reply:
Well, that's life. Maybe some other Animal types are too strong to be bitten :)

For now you should check if the target is the instance of herbivoire.